### ISSUE TITLE
Support for enterprise architectures - ppc64le and s390x

### ISSUE THREAD

**Issue by bencz** [arangodb/arangodb#18001]
I would like to know what is the complexity of porting arangodb to work in enterprise architectures like ppc64le (powerpc64 little endian) and s390x (remembering that s390x is big endian) ?

I saw some points regarding the use of vector instructions and extensions such as SIMD, SSE and AVX...
There is a very interesting project that can be used to transform these calls to an agnostic model that works on multiple architectures, such as this project: 
* https://github.com/berenger-eu/inastemp
* https://github.com/simd-everywhere/simde

**Comment by Dronplane** [comment]
Hi. 
It's hard to quickly give exact estimation. In general codebase is not tied intentionally to any particular architecture.  Some arch-dependent optimizations exist, but they could be disabled in most cases. But there are third-party components (like V8) that also should support targeted architecture. With ARM it was already supported, so all went more less smoothly.  With enterprise architectures that might be a problem.

**Comment by bencz** [comment]
@Dronplane thanks!
Well, IBM give support on V8 to support s390x and ppc64le archs...
I just give a try to build arango on ppc64le and s390x and i got some errors on cmake, about not supported arch...
Anyway, maybe the biggest problem could be the big endian on s390x

**Comment by konsultaner** [comment]
@Dronplane do you see a chance to build it to WebAssembly?

**Comment by dothebart** [comment]
No. Converting C++ / Asembler code to webassembly wouldn't make much sense. 

In other parts of the topic, as we discussed https://github.com/arangodb/arangodb-docker/issues/53 a Port never makes too much sense without suPort. Therefore CI resources etc. would need to be available, machines to run the tests on, and so forth. 
Barely making it compile usually is only half of the game, without CI this tends to quickly detoriate. Hence a Port demands dedication, Machine and Human resources to do. 
So in the end it probably boils down to discuss the commercial aspects off all this first. https://www.arangodb.com/contact/ could probably be used best for that.

**Comment by konsultaner** [comment]
@dothebart My question was probably asked in the wrong way. I just wanted to know if it is even possible in general. I'm currently experimenting with WebAssembly and was just wondering, if this was worth experimenting on.

**Comment by dothebart** [comment]
ah - so this is basically unrelated to this ticket? ;-)

@jvolmer and @markuspf did something about wasm  in conjunction with Pregel / Data science library:
https://github.com/arangodb/arangodb/pull/15876
I don't know whether there is any user documentation about this and whats the current state, maybe they can tell you.

**Comment by konsultaner** [comment]
@dothebart yes sorry, off topic. Thx anyways!
